Automattic/wp-calypso
# [ETK] Update `@wordpress/interface` to the latest published version

Update it to match the changes introduced by the same package in Gutenberg 15.2.

**IMPORTANT: This PR should be merged before releasing Gutenberg 15.2 to WPCOM.**

The mismatch between versions is causing a WSOD in simple sites (and probably in AT, too). Gutenberg 15.2 introduced some changes in this package (https://github.com/WordPress/gutenberg/pull/47498), but ETK ends up loading an older version without these changes. The ETK version is probably loaded (enqueued) after the GB version and then replaces most of it*, while the GB version still tries to import from the replaced module `selectors.js` and fails, since it contains the old code.

_*only some parts, probably because the package is code-split and ETK only loads some modules (like selectors.js)? I haven't looked deeply, this is just an assumption._

Related to https://github.com/WordPress/gutenberg/pull/47498.

Slack discussion: p1677020860615489-slack-CBTN58FTJ (also a good primer on why the `.yarmlrc.yml` changes were needed)

Sum up on Slack: p1677778450155879/1677777680.547219-slack-CBTN58FTJ

## Proposed Changes

Update the version of `@wordpress/interface` in ETK to match the one in Gutenberg 15.2.

## Testing Instructions

1. Apply this diff: D102317-code
to your sandbox, make sure your test sandboxed site has the `gutenberg-edge` sticker applied;
2. Try to create a new post
3. You'll get a WSOD
4. Sync a custom ETK build from this branch (`yarn && yarn dev --sync`)
5. Reload the page
6. It should load the editor without any fatal errors.
